pkg/agent/controller/egress/egress_controller.go

```diff
@@ -521,17 +522,27 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
-func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, nodeName string) error {
-	if egress.Status.EgressNode == nodeName {
+func (c *EgressController) updateEgressStatus(egressName, nodeName string) error {
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		egress, err := c.crdClient.CrdV1alpha2().Egresses().Get(context.TODO(), egressName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		actualStatus := egress.Status.EgressNode
+		if actualStatus == nodeName || (nodeName == "" && actualStatus != c.nodeName) {
```

Comment:
This special check worths a comment. otherwise what the method does may be surprizing.

Reply:
done

Reply:
How about changing the argument to a bool `isLocal` to make the method more readable?
If it's true, the method needs to ensure `egress.Status.EgressNode = c.nodeName`.
If it's false, the method needs to ensure `egress.Status.EgressNode != c.nodeName` by updating it to `""`.